addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -106,21 +107,27 @@ var SlideUploadDialog = Dialog.extend({
      */
     _bindSelect2Dropdown: function () {
         var self = this;
-        this.$('#category_id').select2(this._select2Wrapper(_t('Section'), false,
-            function () {
-                return self.rpc('/slides/category/search_read', {
-                    fields: ['name'],
-                    domain: [['channel_id', '=', self.channelID]],
-                });
-            })
-        );
+        this.$('#category_id').select2(this._select2Wrapper(_t('Section'), false, this._fetchSections.bind(this)));
         this.$('#tag_ids').select2(this._select2Wrapper(_t('Tags'), true, function () {
             return self.rpc('/slides/tag/search_read', {
                 fields: ['name'],
                 domain: [],
             });
         }));
     },
+    /**
+     * Fetch the sections
+     * @private
+     */
+    _fetchSections: async function () {
+        if (!this.values) {
```

Comment:
Please do not use generic variable names such as "values", "list" or "data".

Sugg: "this.slideCategories".